src/io.rs

```diff
@@ -0,0 +1,94 @@
+//! A module for writing TFRecords, Tensorflow's preferred on-disk data format.
+//!
+//! See the [tensorflow docs](https://www.tensorflow.org/api_guides/python/python_io#tfrecords-format-details) for details of this format.
+
+extern crate byteorder;
+extern crate crc32c;
+
+use std::io;
+use std::io::prelude::*;
+
+use self::byteorder::WriteBytesExt;
+
+/// A type for writing bytes in the TFRecords format.
+#[derive(Debug)]
+pub struct RecordWriter<W: Write> {
+    writer: W
+}
+
+impl<W> RecordWriter<W> where W: Write {
+    /// Construct a new RecordWriter which writes to `writer`.
+    pub fn new(writer: W) -> Self {
+        RecordWriter { writer }
+    }
+
+    /// Write a complete TFRecord.
+    pub fn write_record(&mut self, bytes: &[u8]) -> io::Result<()> {
+        /* A TFRecords file contains a sequence of strings 
+        with CRC32C (32-bit CRC using the Castagnoli polynomial) hashes. Each record has the format
+
+        uint64 length
+        uint32 masked_crc32_of_length
+        byte   data[length]
+        uint32 masked_crc32_of_data
+        and the records are concatenated together to produce the file. CRCs are described here [1], and the mask of a CRC is
+        [1] https://en.wikipedia.org/wiki/Cyclic_redundancy_check
+        masked_crc = ((crc >> 15) | (crc << 17)) + 0xa282ead8ul
+        */
+        let mut len_bytes = vec![];
```

Comment:
Could you change `len_bytes`, `len_crc32c_bytes`, and `bytes_cr32_bytes` from vectors to arrays; i.e. `[u8; 8]` and `[u8; 4]`?  That would avoid heap allocations, since they're small and fixed-size.

Reply:
Good catch, thank you. Done.